README.md

```diff
@@ -308,7 +269,7 @@ It supports formatters (including custom ones) that you can use to specify the f
  
 ```ruby
 Gruf.configure do |c|
-  c.instrumentation_options[:request_logging] = {
+  c.interceptors[Gruf::Interceptors::Instrumentation::RequestLogging::Interceptor] = {
```

Comment:
This style of registering interceptors feels like it gives a little too much power to the consumers. From this (I haven't looked at the implementation yet), I think is probably using a hash to guarantee uniqueness. The problem with this approach is when interceptor order becomes important.

I think we can get more flexibility to change and new features by following a process like Rack::Middleware does:
```
c.interceptors.use Gruf::Interceptors::Instrumentation::RequestLogging::Interceptor, options: { formatter: :logstash}
```

Then, in the future, we can add the ability to control order without changing the underlying data structure via something like:
```
c.interceptors.insert_before ...
c.interceptors.insert_after ...
```

What do you think?

Reply:
I thought about this. So in Ruby insertion order is maintained in Hash ([see here](https://www.igvita.com/2009/02/04/ruby-19-internals-ordered-hash/)).

I understand the idea here though, and agree with you that implicit behavior can be a bit confusing. I'll look into this.

Reply:
Awesome! Thanks for providing a link on hash insertion order and looking into changing the API.